aten/src/ATen/native/cuda/ForeachMinMaxFunctors.cuh

```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+#include <ATen/NumericUtils.h>
+
+namespace at {
+namespace native {
+
+// std:: does not have clamp functors
+template <typename T>
+struct minimum {
+    __device__ T operator()(const T& a, const T& b) const { return _isnan(a) or a < b ? a : b; }
```

Comment:
how does this work? You have to return `a` if `a` is `nan`, but instead you are returning `true` converted to `T`?

Reply:
This line is intended ot be the shorthand of 

```
if(_isnan(a))
  return a
else
  return a < b ? a : b
```

It's doing `(_isnan(a) or a < b) ? a : b` and not `_isnan(a) or (a < b) ? a : b`.

I'll add parentheses to make it clearer.
I'll also switch to `||`, I just realized that we are in c++ and `or` magically work for some reason (:

Reply:
It does magically work, it's a c++ keyword https://en.cppreference.com/w/cpp/keyword/or, my bad on operator precedence but parens would make it clearer, thanks!

Reply:
Done